src/lib.rs

```diff
@@ -384,67 +439,86 @@ pub fn make_game() -> Result<Game, JsValue> {
     } = get_canvas_info()
         .map_err(|e| JsValue::from(format!("Error initializing canvas: {}", e)))?;
     let ws = WebSocket::new("ws://localhost:3012")?;
-    let game = Game {
-        speed: 2.0,
-        context,
-        width,
-        height,
-        kill_distance: 64.0,
-        local_player_color: Some(Color::random()),
-        players,
-        socket: ws,
+    let wrapper = GameWrapper {
+        game: Arc::new(Mutex::new(Game {
+            speed: 2.0,
+            context,
+            width,
+            height,
+            kill_distance: 64.0,
+            local_player_color: Some(Color::random()),
+            players,
+            socket: ws,
+        })),
     };
 
-    let onmessage_callback = Closure::wrap(Box::new(move |e: MessageEvent| {
-        // Starting with assuming text messages. Can make efficient later (bson?).
-        if let Ok(txt) = e.data().dyn_into::<js_sys::JsString>() {
-            let strng: String = txt.into();
-            let message: Message = match serde_json::from_str(&strng) {
-                Ok(m) => m,
-                Err(e) => {
-                    console_log!("Unable to deserialize {:?} – {:?}", strng, e);
-                    return;
+    {
+        let game = wrapper
+            .game
+            .lock()
+            .expect("Could not get a lock on the game we just created??");
+
+        let game_clone = wrapper.game.clone();
+        let onmessage_callback = Closure::wrap(Box::new(move |e: MessageEvent| {
+            // Starting with assuming text messages. Can make efficient later (bson?).
```

Comment:
For a game like this it may not make that much of a difference, but you could get very compact with your protocol for most things

Reply:
Yeah!

What's the size of packets you can count on? 1.5KB? That's a _huge_ amount of headspace at this point! Who could ever fill up that much space!?